freva-rest/src/tool_api/endpoints.py

```diff
@@ -0,0 +1,45 @@
+from typing import Dict
+
+from fastapi import Body, Depends, HTTPException, Request, status
+from fastapi.responses import JSONResponse, StreamingResponse
+
+from freva_rest.auth import TokenPayload, auth
+from freva_rest.logger import logger
+from freva_rest.rest import app, server_config
+
+from .tool_abstract import ToolAbstract
+
+
+@app.get("/api/tool/overview", tags=["Tool Overview"])
+async def overview() -> JSONResponse:
+    """Get all available tools and their attributes."""
+    # TODO: Implement method
+    tool = ToolAbstract(name="test", 
+                        author="JaneDoe",
+                        version="0.0",
+                        summary="",
+                        added="2024-11-13",
+                        command="fakecommand",
+                        binary="fakebinary",
+                        output_type="data"
+    )
+    output = tool.model_dump()
+    return JSONResponse(content=output)
+
+@app.post("/api/tool/add", 
```

Comment:
what do u mean  by `add`? submit the job you meant? `Add a tool to the MongoDB` for what? did u considered auth?

Reply:
Add the tool to the MongoDB is the idea here. See `freva-rest/src/freva_rest/tool_api/endpoints.py` for an updated version of this. The idea is that this is the endpoint to post the TOML config of the tool to. Validation of the TOML file would be done server-side and depending if the user is authenticated and the TOML config is valid, endpoint would return a 202 accepted message or one of a list of errors.